add tests for MyButton.tsx

GGalya1 opened the pull request:
- Ensure MyButton renders correctly with provided text
- Verify onClick handler is called on mouse click
- Prevent onClick execution when button is disabled
- Validate additional className is applied correctly
- Test keyboard accessibility (Enter & Space trigger onClick)
- Check button rendering as disabled when prop is passed

GGalya1 committed:
add comprehensive unit tests for button component

- Ensure MyButton renders correctly with provided text
- Verify onClick handler is called on mouse click
- Prevent onClick execution when button is disabled
- Validate additional className is applied correctly
- Test keyboard accessibility (Enter & Space trigger onClick)
- Check button rendering as disabled when prop is passed

Christina-Ev commented on tests/Frontend_Tests/MyButton.test.tsx:
```diff
@@ -0,0 +1,79 @@
+    const button = screen.getByRole("button");
+
+    expect(button).toHaveClass("myBtn");
+    expect(button).toHaveClass("extra-class");
```
I don't think it make makes much of a difference, but you could write:
expect(button.className).toContain("extra-class");

instead of 
expect(button).toHaveClass("extra-class");

GGalya1 replied:
done. See the last commit

Christina-Ev commented on tests/Frontend_Tests/MyButton.test.tsx:
```diff
@@ -0,0 +1,79 @@
+    expect(button).not.toBeDisabled();
+
+    // Focus the button (it is important for Enter/Space. Otherwise it will not be registered)
+    button.focus();
```
You could add below :
button.focus();

the commands:
  await waitFor(() => {
    expect(button).toHaveFocus();
   });
to check if the button is actually focused before sending the key events

Christina-Ev commented on tests/Frontend_Tests/MyButton.test.tsx:
```diff
@@ -0,0 +1,79 @@
+    button.focus();
+
+    // Simulate keys
+    await user.keyboard("{Enter}");
```
below that line you can add:

await waitFor(() => expect(mockOnClick).toHaveBeenCalledTimes(1));
to ensure that the event has been processed before proceeding to the next assertion

Christina-Ev commented on tests/Frontend_Tests/MyButton.test.tsx:
```diff
@@ -0,0 +1,79 @@
+
+    // Simulate keys
+    await user.keyboard("{Enter}");
+    await user.keyboard(" ");
```
Below this line you could add:
  await waitFor(() => expect(mockOnClick).toHaveBeenCalledTimes(2));
for the same reason as the line above

Christina-Ev reviewed (commented):
The test worked fine for me and there was 100% coverage, but I think these small changes may improve it a bit

GGalya1 committed:
test(MyButton): improve className assertion and keyboard interaction tests

- Replaced `toHaveClass` with `toContain` for better class name validation
- Added focus assertion before simulating key events to ensure proper event handling
- Wrapped `expect(mockOnClick).toHaveBeenCalledTimes(1)` inside `waitFor()` to handle async updates
